apps/gallery/test/marionette/fullscreen_image_test.js

```diff
@@ -99,4 +99,50 @@ marionette('the gallery', function() {
       fullscreen_view.fullscreenFrame3);
     assert.strictEqual(translateX, 0);
   });
+
+  test('Change orientation of displayed image in fullscreen', function() {
+    app.tapFirstThumbnail();
+    client.waitFor(function() {
+      return fullscreen_view.displayed;
+    });
+
+    var screen_width = fullscreen_view.screenWidth;
+    var screen_height = fullscreen_view.screenHeight;
+
+    // Checks the image blob is currently being displayed
+    assert.ok(fullscreen_view.hasSrcImageBlobURL(Gallery.ORIGIN,
+      fullscreen_view.displayedImage));
+
+    // Check that there are 5 options displayed beneath the picture
+    assert.equal(fullscreen_view.toolBarOptions.length, 5);
+
+    // Verify that the screen orientation is in portrait mode
+    assert.equal(fullscreen_view.screenOrientation, 'portrait');
+    assert.ok(fullscreen_view.fullScreenToolBar.displayed);
+    assert.equal(fullscreen_view.toolBarWidth, screen_width);
+
+    // Change the screen orientation to landscape mode and verify that
+    // the screen is in landscape mode
+    fullscreen_view.changeOrientation('landscape');
+    assert.equal(fullscreen_view.screenOrientation, 'landscape');
+    assert.ok(fullscreen_view.fullScreenToolBar.displayed);
+    assert.equal(fullscreen_view.toolBarWidth, screen_height);
+
+    // Unlock the screen so that it can be changed back to portrait mode
+    client.executeScript(function () {
+      window.wrappedJSObject.screen.mozUnlockOrientation();
+    });
```

Comment:
That's correct, I tested  by removing the unlock and gallery screen changes to portrait successfully on fullscreen_view.changeOrientation('portrait').  I will update patch to keep mozUnlockOrientation() only while exiting test to release all previous screen locks set by the test.
https://developer.mozilla.org/en-US/docs/Web/API/Screen/unlockOrientation